components/cluster.py

```diff
@@ -99,26 +99,80 @@ def get_component(session_id):
         maxdate = maxdate + datetime.timedelta(days=1)
     items = []
     items.append(html.H1("Clustering"))
-    items.append(html.H3("Select a date range"))
+    items.append(html.H4("Compare clusters"))
+    tab1 = build_cluster_component(mindate, maxdate, session_id, "1")
+    tab2 = build_cluster_component(mindate, maxdate, session_id, "2")
     items.append(
-        html.Div([
-            date_picker.get_component(mindate, maxdate, None),
-            html.Div(
-                id='intermediate-value',
-                style={'display': 'none'},
-                children=[session_id]
-            ),
-            html.Div(id='output-container-date-picker-range')
+        dcc.Tabs([
+            dcc.Tab(label='Date 1', children=[tab1]),
+            dcc.Tab(label="Date 2", children=[tab2])
         ])
     )
     return html.Div(items)
 
 
+def build_cluster_component(mindate, maxdate, session_id, id="1"):
+
+    return html.Div([dcc.DatePickerRange(
+        id='date-pick-'+id,
+        min_date_allowed=mindate,
+        max_date_allowed=maxdate,
+    ),
+        html.Div(
+        id='intermediate-value',
+        style={'display': 'none'},
+        children=[session_id]
+    ),
+
+        html.Div(id='output-clustering-'+id)
+
+    ])
+
+
 @app.callback(
-    dash.dependencies.Output('output-container-date-picker-range', 'children'),
+    dash.dependencies.Output('output-clustering-1', 'children'),
     [
-        dash.dependencies.Input('date-pick', 'start_date'),
-        dash.dependencies.Input('date-pick', 'end_date'),
+        dash.dependencies.Input('date-pick-1', 'start_date'),
+        dash.dependencies.Input('date-pick-1', 'end_date'),
+        dash.dependencies.Input('session-id', "children")
```

Comment:
A minor issue, but  elsewhere in the code we've used `from dash.dependencies import Input, Output, State` so it's possible to use `Output`, `Input`, and `State` without always having to repeat the dependency. Makes the code a bit easier to read